### ISSUE TITLE
reCAPTCHA detected bot actions

### ISSUE THREAD

**Issue by shuqiyg** [cubicbyte/reddit-account-generator#31]
I'm using the default proxies setting but it seems like chrome is able to detect the automation behavior on reCAPTCHA, any suggestion? thanks a ton in advance.

**Comment by shuqiyg** [comment]
I've turn on the VPN on my machine btw. Or maybe me I need to find other proxies? Please advise. Thanks

**Comment by cubicbyte** [comment]
Hi!

If you have recaptcha error, it means that there are too many solve requests coming from your IP (your proxy). Maybe the "cleanliness" of the proxy also matters, but I can't confirm

Tip: paid proxies are good, but not all of them will work good. Residential proxies are the best, but they're very expensive. Free ones will be hard to find, but real

**Comment by shuqiyg** [comment]
> Hi!
> 
> If you have recaptcha error, it means that there are too many solve requests coming from your IP (your proxy). Maybe the "cleanliness" of the proxy also matters, but I can't confirm
> 
> Tip: paid proxies are good, but not all of them will work good. Residential proxies are the best, but they're very expensive. Free ones will be hard to find, but real

Hi there, I've been switching different IPs using VPN, but each IP is only able to create 1-2 accounts then gets flagged by reddit, even I have set a 10 mins timeout.... any way that I can scale it?

**Comment by cubicbyte** [comment]
Reddit has a very strict bot protection system. If 2-3 accounts registered from the same IP, you can have those accounts banned pretty soon, even from your home ip (don't repeat my mistakes 🥲)

So, change IP strictly after each account, otherwise you will get restricted

**Comment by shuqiyg** [comment]
I see, how do I add proxies and where can modify the code of changing IP after each account? Thanks

**Comment by cubicbyte** [comment]
You don't need code, everything happens in the proxies.txt file. Script will use each proxy in the file in order for each account

You can use either a rotating proxy (these are the proxies that will change themselves each time you register) or many static proxies. Put all of them in the proxies.txt file

Static proxies: add the whole list of proxies to the file
Rotating proxy: in this case there will be 1 proxy entry, add it to the file too

**Comment by shuqiyg** [comment]
![image](https://github.com/user-attachments/assets/419a1a0f-8d68-45e9-9952-b069051737f1)
So is this the same to add static proxies? And what about rotating proxy though? 
Thank you for you quick response and really appreciate your answer!

**Comment by cubicbyte** [comment]
Yes, your screenshot shows how to add regular proxies. For rotating proxy it's enough to use only one entry, for example user:pass@144.126.215.152:5258. But note that there are static proxies that require a request to their server to change IP, these proxies will not work

**Comment by shuqiyg** [comment]
I've been changing VPN IPs while using different proxies, reddit seems only allow you to register new account every 10 min, and after a couple accounts been created, the VPN IP will be blocked....any solution?

**Comment by cubicbyte** [comment]
Well, there are inconsistencies in what you're saying. You don't need a vpn, you just need proxies

If reddit limits you to 1 account per 10 minutes, then you're registering accounts from the same IP, means your proxies don't change IPs or you're doing something wrong, can't tell

**Comment by shuqiyg** [comment]
I guess the proxies ain't working then(but the LOG INFO shows the script's been trying to connect through those proxies one after another thought....), it's just me manually rotating IPs from my VPN when it's blocked by reddit. Also, some of those accounts are shadow-banned after making the first vote and comment, any idea why's that?

**Comment by cubicbyte** [comment]
Yea, bans is a problem. Reddit has a very advanced anti-bot system and these accounts aren't really suitable for commenting

Although you can make upvotes from these accounts, but if you do too many upvotes on one post, reddit will ban all accounts at once, so you have to work around it to make it work

**Comment by shuqiyg** [comment]
so only upvote and no comment? 
All of a sudden I got this error today, I didn't modify any code, please help:
![image](https://github.com/user-attachments/assets/820ffce2-056c-4db7-b948-8d2c9b3b9dae)

**Comment by cubicbyte** [comment]
Tbh it's a first time I see this error and I have no clue what it means, can't help. Maybe try to reinstall?

And yes, accounts from this script is mostly suitable for upvote manipulations, not comments

**Comment by im-kent** [comment]
> Tbh it's a first time I see this error and I have no clue what it means, can't help. Maybe try to reinstall?
> 
> And yes, accounts from this script is mostly suitable for upvote manipulations, not comments

This information is useful. Are the accounts suitable for DM? Are user agents spoofed? For the same account, if it is logged in from IPs of different countries every 10 mins or so, is that a problem?

**Comment by cubicbyte** [comment]
To be honest I don't know and would be happy to know as well

**Comment by im-kent** [comment]
> To be honest I don't know and would be happy to know as well

Thanks for your response. Is there a recommended proxy provider that you use? Mine is getting expensive. I don't usually do webdriver stuff, so a ton of junk is eating my bandwidth.

**Comment by cubicbyte** [comment]
Yeah, that's a problem

I used residential proxy but they're really expensive for this thing because of the amount of media on the site. There's no other ways for now if you want a decent accounts, unfortunately

**Comment by lozzsi** [comment]
@im-kent did you make any progress on your questions here "This information is useful. Are the accounts suitable for DM? Are user agents spoofed? For the same account, if it is logged in from IPs of different countries every 10 mins or so, is that a problem?"

**Comment by im-kent** [comment]
> @im-kent did you make any progress on your questions here "This information is useful. Are the accounts suitable for DM? Are user agents spoofed? For the same account, if it is logged in from IPs of different countries every 10 mins or so, is that a problem?"

I did not. It appears that for the time being, Reddit has updated their login logic slightly such that the account generator may not work out of the box (see #32 ). As such I currently am not creating accounts in an automated fashion. I suspect it is probably an easy fix, but I haven't spent any time in the codebase to look into it.

**Comment by lozzsi** [comment]
> I did not. It appears that for the time being, Reddit has updated their login logic slightly such that the account generator may not work out of the box (see #32 ). As such I currently am not creating accounts in an automated fashion. I suspect it is probably an easy fix, but I haven't spent any time in the codebase to look into it.

noted. thanks.
